### ISSUE TITLE
System.Net.HttpClient.DeleteAsync needs to also accept content parameter like the rest of the methods.

### ISSUE THREAD

**Issue by akshays2112** [dotnet/runtime#41473]
### Problem:

In Spotify API endpoint there is a scenario in which you are issuing a Delete command but the data cannot be in the Query String but has to be in the Body/Payload.

### Suggest this Solution
The solution is for System.Net.HttpClient.DeletAsync to be overloaded to accept a content parameter like the rest of the methods.

**Comment by karelz** [comment]
@akshays2112 did you try to use a workaround: `SendAsync` with `DELETE` and `Content`? Would that work?
I wonder how common pattern this is. If not too much, then a workaround might be enough maybe?

**Comment by ManickaP** [comment]
IMHO sending a content in `DELETE` won't be a very common scenario.
Also, other methods which are not expected to carry a payload do not accept `HttpContent` parameter, e.g.: [`GetAsync`](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L328-L328), [`GetByteArrayAsync`](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L214-L214) and other `Get` helpers.

**Comment by akshays2112** [comment]
I have not tried SendAsync. Will look into trying to use it for a workaround and get back to you. Well, I do not want to comment on how common. All I can tell you is contributing to completing SpotifyApi.NetCore to complete it I ran into this problem. Only 4 endpoints to cover so I thought I would try and ask. Thanks again for the swift response.

**Comment by akshays2112** [comment]
SendAsync does not seem to support HttpContent as a parameter either. :(
Any other suggestions for a workaround.

**Comment by ManickaP** [comment]
`HttpContent` is part of `HttpRequestMessage` (which is passed to `SendAsync`): https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs#L61-L61

**Comment by akshays2112** [comment]
Awesome!!! Link to code is really what I was going to ask for next. Will look at the code.

**Comment by akshays2112** [comment]
Need more time. Got involved in something interesting I started on Twitter and buried in a lot of housework. Will work on it soon.

**Comment by scalablecory** [comment]
I agree this seems rare. I don't know if we need a new API for it, as the current `SendAsync` should work for advanced cases. [The RFC](https://tools.ietf.org/html/rfc7231#section-4.3.5) has a note indicating that although this is not disallowed, some implementations have issues with it:

```
   A payload within a DELETE request message has no defined semantics;
   sending a payload body on a DELETE request might cause some existing
   implementations to reject the request.
```

**Comment by akshays2112** [comment]
@scalablecory makes a lot of sense. Even in all the endpoints I did I never ran into this until the very end.

**Comment by akshays2112** [comment]
All I would like is maybe SendAsync can have an overload that will accept ContentType as a constructor parameter like the regular pattern.

I agree since SendAsync works just fine like @scalablecory said it is up to you whether you want to commoditize some of these uncommon cases and make it super simple for even advanced coders.

I am happy to see @karelz has already added this to v6 milestone.

For me, the beauty of sticking to Microsoft platform for coding, though I can easily code the same thing on UNIX/LINUX always where I started, is the ease of use was Microsoft Technologies/Platforms biggest USP (Unique Selling Point).

**Comment by AraHaan** [comment]
I think there should also be an ``DeleteAsJson`` for when you want the Delete request to have an json serialized body of an object that contains values it should send. I am currently getting bit by this issue myself too.

**Comment by patheikkinen** [comment]
Teltonika's router configuration API [uses](https://developers.teltonika-networks.com/reference/7.7.1/v1/auto-reboot/#delete-auto_reboot-scheduler-config) `DELETE` in a manner that would make developers benefit from this feature.
